yt/config.py

```diff
@@ -67,14 +67,22 @@
 )
 
 
-CONFIG_DIR = os.environ.get(
-    "XDG_CONFIG_HOME", os.path.join(os.path.expanduser("~"), ".config", "yt")
-)
-if not os.path.exists(CONFIG_DIR):
-    try:
-        os.makedirs(CONFIG_DIR)
-    except OSError:
-        warnings.warn("unable to create yt config directory")
+def config_dir():
+    config_root = os.environ.get(
+        "XDG_CONFIG_HOME", os.path.join(os.path.expanduser("~"), ".config")
+    )
+    conf_dir = os.path.join(config_root, "yt")
+
+    if not os.path.exists(conf_dir):
```

Comment:
I'm really not a fan of this feature (silently create the config file). I like it even less as a function, since it allows for repetitive warnings in case anything goes wrong at creation.
If you think there's a good reason (other than historic) to do this, can you please document it in #3045 ?

Reply:
It's not creating config file, but a config directory, which is pretty much a default behavior for a lot of software:

* https://github.com/matplotlib/matplotlib/blob/master/lib/matplotlib/__init__.py#L419-L422

Also noting the most important thing: it was already there and changing config related behavior is not the point of this PR.

Reply:
alright I'll open a different PR to fix this